Enhance: Add version release type Splash-screens

SlySven opened the pull request:
Also removes obsolete old un-extension-ed Splash-screen file from repo.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
W.I.P. Add version release type Splash-screens (release/PTB/development)

Image files for PTB and Development are not finalised yet.

Also removes obsolete old un-extension-ed Splash-screen file from repo.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use original MUDLET text logo for PTB splash-screen

An .svg original was not immediately available but I manged to regenerate
something from another .png artwork. The quality is not brilliant but it
is, at least, the right shape.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

demonnic commented:
files changed seems to delete the main splashscreen file?

vadi2 commented:
That was a duplicate, as it turns out - `Mudlet_splashscreen_main.png` is still there.

demonnic reviewed (approved):
Well in that case

vadi2 commented on src/mudlet.qrc:
```diff
@@ -178,7 +178,9 @@
         <file>mudlet-mapper.xml</file>
+        <file>Mudlet_splashscreen_development.png</file>
         <file>Mudlet_splashscreen_main.png</file>
+        <file>Mudlet_splashscreen_ptb.png</file>
```
The `Mudlet` text rendering doesn't look as good as what we have on the main one - could that be improved?

Also, the sunlight spilling into the arch - could you lower it just a tad?

I think this will make it look even better.

SlySven replied:
That was supposed to be a MOON not a Sun... yeah, I'll see if I can turn it down a bit! :sweat_smile:

SlySven replied:
I'll have another go - I see now that the process I used to extract the original logo text uses a binary yes/no to select the pixels that made up the text - but that falls over when the text is anti-aliased as it is here. Thorsten did get back to me and he confirmed what you said that the text was constructed by hand - he express surprise that we did not have the .svg file for it but the only file of that type we have is just for the arch itself - NOT for the text.  I have explained this in a reply but have not had another response yet.  Unless you happen to have it tucked away somewhere?

vadi2 replied:
We do have the svg's! You should have asked. See all of the svg's in the https://github.com/Mudlet/artwork repo :)

vadi2 commented on src/mudlet.qrc:
```diff
@@ -178,7 +178,9 @@
         <file>mudlet-lua/lua/generic-mapper/generic_mapper.xml</file>
         <file>mudlet-lua/lua/utf8_filenames.lua</file>
         <file>mudlet-mapper.xml</file>
+        <file>Mudlet_splashscreen_development.png</file>
```
It looks really rather "broken" - and from what I remember this was exactly the idea you're trying to convey - but it's not appealing to be using something that already looks broken from the get-go.

SlySven replied:
Not broken, more a *work-in-progress* like the text hasn't been painted yet (but the can of paint is there waiting, resting on the hammer that will be needed  to put the last two letters up). The partially bordered up exit with the hazard tape is to reinforce that this version may not work properly and is not a finished release version.

Given that the other one is labelled with "PTB" would adding "DEV" in the same format help?

I had an alternative design in mind which I thought I prepared a long while back but cannot find which involved a really red sun just rising over the horizon (much lower than the moon in the PTB, and with a much stronger flare effect) and a strong red tint to the whole image (like something like a "vampire's eye" view in a paranormal fantasy film) - sort of indicating in a (not) subtle way that this version is working towards the *new* day of a Mudlet release - would that be preferable do you think?

As it is I do not think we should keep development versions (having something in APP_BUILD other than something beginning with `-public-test-build`) having the same splash screen as the release version.  Though we ought then remind packagers that if they follow our advice to add something to that variable to identify their version as a packaged one then either they will need to patch the splashscreen selection code in `main.cpp` and `dlgAboutDialog.cpp` OR we add ***another*** case that allows "branding"...

vadi2 replied:
> would that be preferable do you think?

I think that sounds good.

SlySven committed:
Revise: tweak the development and ptb splash-screen images

Also add a working copy of the GIMP native file that I was using to create
the splash-screens so that it is on record.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap commented:
I am not sure how I like the different fonts and the boxes around individual letters. 

Can't we simpley write "DEV" and "PTB" in the same style as "Mudlet"?

Like the images a lot actually. Much better than the first draft.

SlySven commented:
No, because the original "Mudlet" uses glyphs that Thorwill created and the alphabet only contains `M`, `d`, `e`, `l`, `t`,  and `u`! :slightly_smiling_face:

Kebap commented:
I made a thing: 
![image](https://user-images.githubusercontent.com/117238/77794599-7c280d80-706c-11ea-92ff-a6409bbf8557.png)
